### PR TITLE
Cooler hivebots

### PR DESCRIPTION
**Pull request by nemvar**
## About The Pull Request

Gives hivebots improved sprites. The one that changed the most is the strong hivebot, who now has two eye slids instead of one, making them look less similar to the normal hivebots. 

All of them have different eye colors when aggroed (red) and when idle (green). Their 'eye' is now moving from left to right, faster when they have a target. They sometimes spout some stupid taunts when aggroing a mob. When angery, they emit a light.

Hivebots now have the standard bot verbs and text span.

As for gameplay related changes, hivebots now go into harm intent when they have a target, meaning you can no longer run through them. Player controlled hivebots get a verb to toggle their sprite and intent.

Coolhivebots
coolhivebots2

The eye thing is moving now, that's why it's not visible on the upper picture.

## Why It's Good For The Game

These things felt (and still feel) very bland, but now less so.

## Changelog
:cl:
tweak: Various cosmetic and gameplay related changes to hivebots. Player controlled hivebots can change their look by changing their intent to harm.
/:cl:

### CODE REVIEW

**Comment by AarontheIdiot** [comment]
can we get some images

**Comment by Detective-Google** [comment]
Nice

**Comment by Tlaltecuhtli** [comment]
give them animated sprites tbh

**Comment by habader** [comment]
when are we getting deep hivebot lore

**Comment by chaoko99** [comment]
I find it hilarious because these are the old sprites that TG used YEARS ago for hivebots.

File: code/modules/mob/living/simple_animal/hostile/hivebot.dm [hunk 1]
```diff
@@ -35,6 +40,35 @@
...
+	if(a_intent != INTENT_HELP)
+		a_intent = INTENT_HELP
+	else
+		a_intent = INTENT_HARM
+	update_icons()
+
+/mob/living/simple_animal/hostile/hivebot/update_icons()
+	if(a_intent != INTENT_HELP)
+		icon_state = "[initial(icon_state)]_attack"
+		set_light(6, 0.4, COLOR_RED_LIGHT)
```
Comment (ShizCalev) [inline code/modules/mob/living/simple_animal/hostile/hivebot.dm]: use the mob_light proc instead.

File: code/modules/mob/living/simple_animal/hostile/hivebot.dm [hunk 2]
```diff
@@ -35,6 +40,35 @@
...
+	else
+		icon_state = initial(icon_state)
+		set_light(0)
```
Comment (ShizCalev) [inline code/modules/mob/living/simple_animal/hostile/hivebot.dm]: .

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by PKPenguin321** [comment]
spritework request: different colors for different hivebot types would be good for better visual feedback

**Comment by AnturK** [comment]
I don't like the special intent handling here that should be in generic way at simple mob or even mob level.

**Comment by optimumtact** [comment]
stop making simple bots non simple please, if you want to add lots of features you have to graduate them to mob/carbon or other typepaths

**Comment by nemvar** [comment]
I got rid of the snowflake intent handling. I don't quite get the complaint about making simplemobs too complex. These things are still as basic as it can get.

File: code/modules/mob/living/simple_animal/hostile/hivebot.dm [hunk 3]
```diff
@@ -23,18 +23,47 @@
...
+	. = ..()
+	a_intent_change(INTENT_HARM)
+	if(prob(5))
+		say(pick("INTRUDER DETECTED!", "CODE 7-34.", "101010!!"), forced = type)
+
+/mob/living/simple_animal/hostile/hivebot/LoseAggro()
+	. = ..()
+	a_intent_change(INTENT_HELP)
+
+/mob/living/simple_animal/hostile/hivebot/a_intent_change(input as text)
```
Comment (AnturK) [inline code/modules/mob/living/simple_animal/hostile/hivebot.dm]: All good now but does intent screen object show by default for simple animals ?

Comment (nemvar) [reply code/modules/mob/living/simple_animal/hostile/hivebot.dm]: No, you have to type out the verb

File: code/modules/mob/living/simple_animal/hostile/hivebot.dm [hunk 4]
```diff
@@ -23,18 +23,47 @@
...
+	. = ..()
+	update_icons()
+
+/mob/living/simple_animal/hostile/hivebot/update_icons()
+	if(a_intent != INTENT_HELP)
+		icon_state = "[initial(icon_state)]_attack"
+		alert_light = mob_light(COLOR_RED_LIGHT, 6, 0.4)
```
Comment (ShizCalev) [inline code/modules/mob/living/simple_animal/hostile/hivebot.dm]: make sure you're not leaving an old alert light stranded.

Comment (nemvar) [reply code/modules/mob/living/simple_animal/hostile/hivebot.dm]: done

File: code/modules/mob/living/simple_animal/hostile/hivebot.dm [hunk 5]
```diff
@@ -23,18 +23,47 @@
...
+/mob/living/simple_animal/hostile/hivebot/LoseAggro()
+	. = ..()
+	a_intent_change(INTENT_HELP)
+
+/mob/living/simple_animal/hostile/hivebot/a_intent_change(input as text)
+	. = ..()
+	update_icons()
+
+/mob/living/simple_animal/hostile/hivebot/update_icons()
+	qdel(alert_light)
```
Comment (AnturK) [inline code/modules/mob/living/simple_animal/hostile/hivebot.dm]: QDEL_NULL, it won't get cleanly deleted if you change to help intent.

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/hostile/hivebot.dm (hunk 5 above)
Comment (nemvar) [reply code/modules/mob/living/simple_animal/hostile/hivebot.dm]: Did that. May I ask why it's needed?

**Pull request merged**